render.ts

```diff
@@ -74,11 +78,31 @@ export function process(rawDetails: RawInput): ProcessedInput {
 }
 
 function processEntry(raw: RawEntry): ProcessedEntry {
+    // Set some default in case those values aren't specified
+    if (raw.km == undefined) {
+        raw.km = 0;
+    }
+    if (raw.start_time == undefined) {
+        raw.start_time = "00:00";
+    }
+    if (raw.end_time == undefined) {
+        raw.end_time = "24:00";
+    }
+
+    // Calculate the hours based on the given time strings.
+    // Let the javascript dateparser do its job.
+    var start = new Date(`01/01/2024 ${raw.start_time}`);
+    var end = new Date(`01/01/2024 ${raw.end_time}`);
+
+    var milliseconds_diff = end - start;
+    var hours = Math.floor(milliseconds_diff / 1000 / 60 / 60);
+    raw.hours = raw.hours ?? hours
+
     let food_money = 0;
-    if (raw.hours >= 8) {
-        food_money = 14;
-    } else if (raw.hours > 24) {
-        food_money = 24;
+    if (raw.hours >= 8 && raw.hours < 24) {
+        food_money = SMALL_FOOD_MONEY;
+    } else if (raw.hours >= 24) {
+        food_money = BIG_FOOD_MONEY;
     }
```

Comment:
That logic was actually pretty buggy before and wouldn't trigger.
Also the `24` was wrong, it's `28`.

Reply:
Huh, may have been an `undefined`/`null` problem. Thanks for fixing it :)

Reply:
(and oof to the 24 bug, so much money lost)

Reply:
Oh no, it was because of the `> 24` hours check, wow :facepalm:

Reply:
Both `> 24` and before it only checked `>=8`, which is also true for `24`, so it captured early. :D